test/ocpbugs48050/server.go

```diff
@@ -0,0 +1,217 @@
+	writer := bufio.NewWriter(conn)
+
+	// Read the request line (e.g., "GET /duplicate-te HTTP/1.1").
+	requestLine, err := reader.ReadString('\n')
+	if err != nil {
+		log.Printf("%v: Error reading request line: %v\n", conn.LocalAddr(), err)
+		return
+	}
+
+	log.Printf("%v: Received request: %s", conn.LocalAddr(), requestLine)
+
+	// Split the request line to get the method, path, and protocol.
+	var method, path, protocol string
+
+	if _, err = fmt.Sscanf(requestLine, "%s %s %s", &method, &path, &protocol); err != nil {
+		log.Printf("%v: Error parsing request line: %v\n", conn.LocalAddr(), err)
+		return
+	}
+
+	switch path {
+	case "/healthz":
+		healthResponse := "HTTP/1.1 200 OK\r\n" +
+			"Content-Type: text/plain\r\n" +
+			"Connection: close\r\n\r\n" +
+			"OK\r\n"
+		if _, err := writer.WriteString(healthResponse); err != nil {
+			log.Printf("%v: Error writing /healthz response: %v\n", conn.LocalAddr(), err)
+			return
+		}
+
+	case "/single-te":
+		if err := handleSingleTE(writer); err != nil {
+			log.Printf("%v: Error handling /single-te: %v\n", conn.LocalAddr(), err)
+			return
+		}
+
+	case "/duplicate-te":
+		if err := handleDuplicateTE(writer); err != nil {
+			log.Printf("%v: Error handling /duplicate-te: %v\n", conn.LocalAddr(), err)
+			return
+		}
+
+	default:
+		notFoundResponse := "HTTP/1.1 404 Not Found\r\n" +
+			"Content-Type: text/plain\r\n" +
+			"Connection: close\r\n\r\n" +
+			"404 page not found\r\n"
+		if _, err := writer.WriteString(notFoundResponse); err != nil {
+			log.Printf("%v: Error writing 404 response: %v\n", conn.LocalAddr(), err)
+			return
+		}
+	}
+
+	if err := writer.Flush(); err != nil {
+		log.Printf("%v: Error flushing data to connection: %v\n", conn.LocalAddr(), err)
+	}
+}
+
+// handleSingleTE handles the response for the /single-te path.
+func handleSingleTE(writer *bufio.Writer) error {
+	response := fmt.Sprintf("HTTP/1.1 200 OK\r\n"+
+		"Date: %s\r\n"+
+		"Content-Type: text/plain; charset=utf-8\r\n"+
+		"Connection: close\r\n"+
+		"Transfer-Encoding: chunked\r\n\r\n", time.Now().UTC().Format(time.RFC1123))
+
+	if _, err := writer.WriteString(response); err != nil {
+		return fmt.Errorf("error writing single-te response headers: %v", err)
+	}
+
+	// Form the chunked response
+	chunks := []string{
+		// First chunk:
+		// - 'A' is the hexadecimal representation of 10 (the length of "single-te\n").
+		// - \r\n separates the chunk size from the chunk data.
+		// - "single-te\n" is the chunk data (the \n is included in the chunk data).
+		// - \r\n terminates the chunk.
+		"A\r\nsingle-te\n\r\n",
+
+		// Final chunk:
+		// - '0' indicates a chunk of zero length, signaling the end of the chunked message.
+		// - The first \r\n separates the chunk size (0) from what would be the chunk data.
+		// - The second \r\n terminates the zero-length chunk and the entire chunked message.
+		"0\r\n\r\n",
+	}
+
+	for i := range chunks {
+		if _, err := writer.WriteString(chunks[i]); err != nil {
```

Comment:
Will these be real chunks or `WriteString()` will buffer and send as a single chunk? I mean do we need to flush after each write?

Reply:
Yes, I think we should. Pushed a change for flushing per chunk in https://github.com/openshift/cluster-ingress-operator/pull/1150/commits/002057263f278a3aa2198ef26d269b7ef74496c5.